config/riscv64/chips/black-parrot/bsg_newlib_intf.c

```diff
@@ -0,0 +1,27 @@
+#include <stdlib.h>
+#include <machine/bsg_newlib_fs.h>
+
+void bsg_newlib_init(void) {
+
+    // Init file system
+    if(bsg_newlib_fs_init() < 0) {
+      exit(-1);
+    }
+}
+
+void bsg_newlib_exit(int exit_status) {
+  //EXIT_SUCCESS is 0 
+  if(exit_status == EXIT_SUCCESS) {
+    exit(0);
```

Comment:
I guess this is a newlib question, but what does this exit function actually do? Does it write to our ufinish CSR?

Reply:
Yeah, this maps to the syscall.c exit function which writes the the ufinish csr. I tried to move this exit function to where libgloss was but then I had to move the encoding to libgloss to be able to write to the csr so it's all just kept local and confined within the file for now.

Reply:
Works for me